### ISSUE TITLE
Using the language select can cause lost work. 

### ISSUE THREAD

**Issue by mmzeeman** [zotonic/zotonic#1513]
The language select does a page reload. This can cause your work to be lost when you are busy editting a resource.

**Comment by ArthurClemens** [comment]
The same goes for other menu items. The form should get a dirty state and use that to warn when you navigate away.

**Comment by mmzeeman** [comment]
For users it is probably better to let the system solve this problem. 

Is there a method to store page state locally?

**Comment by ArthurClemens** [comment]
A common pattern is to get a warning when you have edited some text in a editor, something like "You have unsaved changes.", then give the opportunity to save or abandon the work.

To implement it, the tricky part is to let the various inputs in the form set the dirty state.

**Comment by mmzeeman** [comment]
Those dialogs thoroughly derail the thought train of users. It is better to automatically remember the state and let users finish the thing they wanted to do and then offer to restore the state when they return.

**Comment by ArthurClemens** [comment]
I agree in general. But the language selection is (unintended) a navigating away like the other menu selections. How would you deal with clicking on other menu items?

**Comment by mmzeeman** [comment]
Thinking out loud now.

Maybe something like this?

Make a hook to `window.unload`... which automatically saves things when the user navigates away.

When a page is loaded, check localstorage to see if there are recent unsaved changes for this page? If so, offer to restore the changes.

**Comment by ddeboer** [comment]
I really like the idea! It would be similar to GitHub, where if you type some text (for instance, in the comment box below), navigate away and come back, the text is still there.

**Comment by ArthurClemens** [comment]
You need to notify the user that that text is not saved yet. Otherwise the user will never click on the Save button.

**Comment by ddeboer** [comment]
Yep, it makes sense to combine both ideas: show a fixed banner ‘Undaved changes’ *and* allow users to navigate away and come back.

**Comment by mmzeeman** [comment]
It is not difficult to check for changes. The value property and the value available via js differ then. We could make a generic library for this.
